### ISSUE TITLE
Uncaught TypeError: Cannot read property 'canInstall' of undefined

### ISSUE THREAD

**Issue by ontwikkelfabriek** [Bartozzz/vue-pwa-install#1]
I have a pwa app (quasar) that installs and works. I have tried to implement your install pwa button. When deploying I get the following error:

'Uncaught TypeError: Cannot read property 'canInstall' of undefined'

```vue
<template>
  <q-page class="flex flex-center">
    <div class="q-pa-md" id="install-banner" v-if="deferredPrompt">
      <q-btn
        @click="promptInstall"
        flat
        color="primary"
        label="Add to Home Screen"
      />
    </div>
  </q-page>
</template>

<script>
export default {
  data() {
    return {
      deferredPrompt: undefined
    };
  },
  methods: {
    promptInstall() {
      // Show the prompt:
      this.deferredPrompt.prompt();

      // Wait for the user to respond to the prompt:
      this.deferredPrompt.userChoice.then(choiceResult => {
        if (choiceResult.outcome === "accepted") {
          console.log("User accepted the install prompt");
        } else {
          console.log("User dismissed the install prompt");
        }

        this.deferredPrompt = null;
      });
    }
  },
  created() {
    this.$on("canInstall", event => {
      // Prevent Chrome 67 and earlier from automatically showing the prompt:
      event.preventDefault();

      // Stash the event so it can be triggered later:
      this.deferredPrompt = event;
    });
  }
};
</script>
```

**Comment by Bartozzz** [comment]
Could you send me the error stack trace? It seems that you are trying to access `canInstall` property of an object that doesn't exists – I believe that it is not related to this library. Also, please make sure that the plugin is registered correctly, as shown in the [Installation steps](https://github.com/Bartozzz/vue-pwa-install#installation).

**Comment by ontwikkelfabriek** [comment]
I did install it correctly i think. 

```
import Vue from "vue";
import VueInstall from "vue-pwa-install";

Vue.use(VueInstall);
```


There's not much of a stacktrace.


```
app.88f9c2a2.js:1 Service worker has been registered.
app.88f9c2a2.js:1 New content is downloading.
33Fetch finished loading: GET "<URL>".
app.88f9c2a2.js:1 Content has been cached for offline use.
68vendor.69cee01a.js:7 Uncaught TypeError: Cannot read property 'canInstall' of undefined
    at S.t.$emit (vendor.69cee01a.js:7)
    at S.installHandler (vendor.69cee01a.js:62)
t.$emit @ vendor.69cee01a.js:7
S.installHandler @ vendor.69cee01a.js:62
```

![pwa-error](https://user-images.githubusercontent.com/9429959/75045293-4e770400-54c3-11ea-8ae6-45f87d0aa90c.png)


I 'solved' it by using  plain js code in de created() hook:

```
window.addEventListener('beforeinstallprompt',  event => {
    console.log("before install prompt")
    this.deferredPrompt = prompt;
    ect...
  })
```

**Comment by Bartozzz** [comment]
In fact, it seems like an error from the Vue source. I'll try to have a look at this later today.

**Comment by rudolfbyker** [comment]
I'm getting the same error when using component-specific mixins:

`main.js`:
```javascript
Vue.use(VueInstall, {
  // Don't add mixin globally.
  mixin: false,
});
```
`MyComponent.vue`:
```javascript
import { VueInstallMixin } from "vue-pwa-install";

export default {
  mixins: [VueInstallMixin],
```

But when using global mixins, it works fine:

`main.js`
```javascript
Vue.use(VueInstall, {
  // Do add mixin globally.
  mixin: true,
});
```
`MyComponent.vue`:
```javascript
export default {
  mixins: [],
```

**Comment by Bartozzz** [comment]
This should be resolved in [`v1.1.0`](https://www.npmjs.com/package/vue-pwa-install). You can have a look at [`.codesandbox/sandbox`](https://github.com/Bartozzz/vue-pwa-install/tree/master/.codesandbox/sandbox) for examples.
